application/src/main/java/org/thingsboard/server/controller/RuleChainController.java

```diff
@@ -114,6 +105,7 @@
 @RestController
 @TbCoreComponent
 @RequestMapping("/api")
+@RequiredArgsConstructor
```

Comment:
This annotation not used. Remove this or make "Autowired" services final

Reply:
Ok